src/HighchartsReact.js

```diff
@@ -1,66 +1,93 @@
-    const constructorType = props.constructorType || "chart";
+const HighchartsReact = forwardRef(function HighchartsReact(props, ref) {
+  const containerRef = useRef();
+  const chartRef = useRef();
+  const unmountingRef = useRef(false);
 
-    if (!H) {
-      console.warn('The "highcharts" property was not passed.');
-    } else if (!H[constructorType]) {
-      console.warn('The "constructorType" property is incorrect or some ' +
-        'required module is not imported.');
-    } else if (!props.options) {
-      console.warn('The "options" property was not passed.');
-    } else {
-      // Create a chart
-      this.chart = H[constructorType](
-        this.container.current,
-        props.options,
-        props.callback ? props.callback : undefined
-      );
-    }
-  }
-
-  componentDidMount() {
-    this.createChart();
-  }
+  useIsomorphicLayoutEffect(() => {
+    return () => {
+      unmountingRef.current = true;
+    };
+  }, []);
 
-  componentDidUpdate() {
-    const props = this.props;
+  useIsomorphicLayoutEffect(() => {
+    function createChart() {
+      const H = props.highcharts || Highcharts;
+      const constructorType = props.constructorType || 'chart';
 
-    if (props.allowChartUpdate !== false) {
-      if (!props.immutable && this.chart) {
-        this.chart.update(
-          props.options,
-          ...(props.updateArgs || [true, true])
+      if (!H) {
+        console.warn('The "highcharts" property was not passed.');
+      } else if (!H[constructorType]) {
+        console.warn(
+          'The "constructorType" property is incorrect or some ' +
+            'required module is not imported.'
         );
+      } else if (!props.options) {
+        console.warn('The "options" property was not passed.');
       } else {
-        this.createChart();
+        // Create a chart
+        chartRef.current = H[constructorType](
+          containerRef.current,
+          props.options,
+          props.callback ? props.callback : undefined
+        );
       }
     }
-  }
 
-  componentWillUnmount() {
-    // Destroy chart
-    if (this.chart) {
-      this.chart.destroy();
-      this.chart = null;
+    if (!chartRef.current) {
+      createChart();
+    } else {
+      if (props.allowChartUpdate !== false) {
+        if (!props.immutable && chartRef.current) {
+          chartRef.current.update(
+            props.options,
+            ...(props.updateArgs || [true, true])
+          );
+        } else {
+          createChart();
+        }
+      }
     }
-  }
 
-  render() {
-    // Create container for the chart
-    return React.createElement(
-      "div",
-      { ...this.props.containerProps, ref: this.container }
-    );
-  }
-}
+    return () => {
+      // Destroy chart only if unmounting.
+      if (chartRef.current && unmountingRef.current) {
+        chartRef.current.destroy();
+        chartRef.current = null;
+      }
+    };
+  });
+
+  useImperativeHandle(
```

Comment:
This was only done for full backwards compatibility with the old API. The empty array ensures this ref is only created once (like it is with class components), and the `get chart()` allows us to simulate a mutable `.chart` instance field. In a future breaking change, it might be better to use ref forwarding and forward the ref to `chartRef` directly, and accept a `containerRef` prop that would be wired up to the `div` instead.